extension/memorylimiterextension/memorylimiter.go

```diff
@@ -0,0 +1,38 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package memorylimiterextension // import "go.opentelemetry.io/collector/extension/memorylimiterextension"
+
+import (
+	"context"
+
+	"go.uber.org/zap"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/internal/memorylimiter"
+)
+
+// newMemoryLimiter returns a new memorylimiter extension.
+func newMemoryLimiter(cfg *Config, logger *zap.Logger) (*memoryLimiterExtension, error) {
+	ml, err := memorylimiter.NewMemoryLimiter(cfg, logger)
+	if err != nil {
+		return nil, err
+	}
+
+	return &memoryLimiterExtension{memLimiter: ml}, nil
+}
+
+func (ml *memoryLimiterExtension) Start(_ context.Context, host component.Host) error {
+	ml.memLimiter.SetBallastSizeIfExtensionExists(host)
+	ml.memLimiter.StartMonitoring()
```

Comment:
Let's replace the two separate methods with `ml.memLimiter.Start(ctx, host)`

Reply:
replaced